src/env/setup_ca.sh

```diff
@@ -1,114 +1,57 @@
+  esac
 done
 
-dnf -y module enable idm:DL1
-dnf -y copr enable jjelen/vsmartcard
-dnf -y install vpcd softhsm python3-pip sssd-tools
-yum groupinstall "Smart Card Support" -y
-
-if [[ $DIR = "" ]]
-then 
-    DIR=.
-fi
-
-SOPIN='12345678'
-PIN='123456'
-export GNUTLS_PIN=$PIN
-VIRT=$DIR
-NSSDB=$VIRT/db
-CONF=$VIRT/conf
-NAME=localuser1
-
-
-if [[ ! -f "$CONF/softhsm2.conf" ]]
-then 
-    echo "File $CONF/softhsm2.conf does not exist"
-    exit 1
-elif [[ ! -f "$CONF/ca.cnf" ]]
-then
-    echo "File $CONF/ca.cnf does not exist"
-    exit 
-elif [[ ! -f "$CONF/req_${NAME}.cnf" ]]
+if [[ -z "$CA_DIR" ]]
 then
-    echo "File $CONF/req_${NAME}.cnf does not exist"
-    exit 1
-elif [[ ! -f "$CONF/virtcacard.cil" ]]
-then
-    echo "File $CONF/virtcacard.cil does not exist"
-    exit 1
-elif [[ ! -f "$CONF/virt_cacard.service" ]]
-then
-    echo "File $CONF/virt_cacard.service does not exist"
-    exit 1
-elif [[ ! -f "$CONF/sssd.conf" ]]
-then
-    echo "File $CONF/sssd.conf does not exist"
-    exit 1
+  err "No working directory provided. Use --dir <path> parameter for specifying CA dir"
 fi
 
-# Configurting softhm2
-P11LIB='/usr/lib64/pkcs11/libsofthsm2.so'
-sed -i "s,<TESTDIR>,$VIRT,g" $CONF/softhsm2.conf
-pushd $VIRT || exit
-
-mkdir tokens
-export SOFTHSM2_CONF="$CONF/softhsm2.conf" # Should I save previous value of 
-softhsm2-util --init-token --slot 0 --label "SC test" --so-pin="$SOPIN" --pin="$PIN"
-
+CONF_DIR="$CA_DIR/conf"
+# Setup local openssl CA
+pushd "$CA_DIR" || exit 1
+log "CA directory $CA_DIR"
 
-# Creating NSS database
-mkdir $NSSDB
-modutil -create -dbdir sql:$NSSDB -force
-modutil -list -dbdir sql:$NSSDB | grep 'library name: p11-kit-proxy.so'
-if [ "$?" = "1" ]; then 
-    modutil -force -add 'SoftHSM PKCS#11' -dbdir sql:$NSSDB -libfile $P11LIB
-fi
+mkdir -p {certs,crl,newcerts}
+log "Directories for local CA are created"
 
-# Setup local openssl CA
-mkdir {certs,crl,newcerts}
 touch serial index.txt crlnumber index.txt.attr
-echo 01 > serial
-openssl genrsa -out rootCA.key 2048
-
-openssl req -batch -config $CONF/ca.cnf -x509 -new -nodes \
-            -key rootCA.key -sha256 -days 10000 -set_serial 0 \
-            -extensions v3_ca -out $VIRT/rootCA.crt
-openssl ca -config $CONF/ca.cnf -gencrl -out crl/root.crl
+echo 01 >serial
+log "Files for local CA are created"
 
-# Setup user and certs on the card
-useradd -m $NAME
-echo -e '654321\n654321' | passwd localuser1
+openssl genrsa -out rootCA.key 2048
+log "Key for local CA is created"
 
-openssl genrsa -out ${NAME}.key 2048
-openssl req -new -nodes -key ${NAME}.key -reqexts req_exts -config $CONF/req_${NAME}.cnf -out ${NAME}.csr
-openssl ca -config $CONF/ca.cnf -batch -notext -keyfile rootCA.key -in ${NAME}.csr -days 365 -extensions usr_cert -out ${NAME}.crt
+openssl req -batch -config "$CONF_DIR"/ca.cnf -x509 -new -nodes \
+  -key rootCA.key -sha256 -days 10000 -set_serial 0 \
+  -extensions v3_ca -out "$CA_DIR"/rootCA.pem
+log "Certificate for local CA is created"
```

Comment:
You can probably generate both with something like 'openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365'

Reply:
yes, will fix this